pyriemann_qiskit/classification.py

```diff
@@ -0,0 +1,317 @@
+        self.feature_dim = nbSensor * nbSamples
+        self.log("Feature dimension = ", self.feature_dim)
+        Xta = X[y == self.target]
+        Xnt = X[np.logical_not(y == self.target)]
+        VectorizedXta = self.vectorize(Xta)
+        VectorizedXnt = self.vectorize(Xnt)
+        self.new_feature_dim = len(VectorizedXta[0])
+        self.log("Feature dimension after vector processing = ",
+                 self.new_feature_dim)
+        return (VectorizedXta, VectorizedXnt)
+
+    def additionnal_setup(self):
+        self.log("There is no additional setup.")
+
+    def fit(self, X, y):
+        self.log("Fitting: ", X.shape)
+        self.prev_fit_params = {"X": X, "y": y}
+        self.classes_ = np.unique(y)
+        VectorizedXta, VectorizedXnt = self.splitTargetAndNonTarget(X, y)
+
+        self.training_input["Target"] = VectorizedXta
+        self.training_input["NonTarget"] = VectorizedXnt
+        self.log(get_feature_dimension(self.training_input))
+        feature_dim = get_feature_dimension(self.training_input)
+        self.feature_map = ZZFeatureMap(feature_dimension=feature_dim, reps=2,
+                                        entanglement='linear')
+        self.additionnal_setup()
+        if self.quantum:
+            if not hasattr(self, "backend"):
+                def filters(device):
+                    return (
+                        device.configuration().n_qubits >= self.new_feature_dim
+                        and not device.configuration().simulator
+                        and device.status().operational)
+                devices = self.provider.backends(filters=filters)
+                try:
+                    self.backend = least_busy(devices)
+                except Exception:
+                    self.log("Devices are all busy. Getting the first one...")
+                    self.backend = devices[0]
+                self.log("Quantum backend = ", self.backend)
+            seed_sim = aqua_globals.random_seed
+            seed_trans = aqua_globals.random_seed
+            self.quantum_instance = QuantumInstance(self.backend, shots=1024,
+                                                    seed_simulator=seed_sim,
+                                                    seed_transpiler=seed_trans)
+        return self
+
+    def get_params(self, deep=True):
+        # Class is re-instanciated for each fold of a cv pipeline.
+        # Deep copy of the original instance is insure trough this method
+        # and the pending one set_params
+        return {
+            "target": self.target,
+            "qAccountToken": self.qAccountToken,
+            "quantum": self.quantum,
+            "processVector": self.processVector,
+            "verbose": self.verbose,
+            "test_input": self.test_input,
+        }
+
+    def set_params(self, **parameters):
+        for parameter, value in parameters.items():
+            setattr(self, parameter, value)
+        return self
```

Comment:
I believe that these functions are already inherited. To check.

Reply:
I made some tests with a validation pipeline and had to implement these methods as a workaround. Could it be the naming convention?

Reply:
Yes, I guess it is exactly related to this comment you gave me:
https://github.com/pyRiemann/pyRiemann/pull/122#discussion_r648021748

Reply:
Ok, I got it sorted by passing test_input as a parameter.
I also provided one other test `test_GetSetParams` using a cross-validation pipeline, to insure cloning is working as expected